core/unit_test/TestReducers.hpp

```diff
@@ -447,6 +447,27 @@ struct TestReducers {
     }
   }
 
+  // Test that reducers return correct results with LaunchBounds value smaller
+  // that 32.
+  static void test_launch_bounds() {
+    Kokkos::View<Scalar*> v("", 31);
+    Kokkos::deep_copy(v, Scalar(10));
+
+    // Functor
+    auto l = KOKKOS_LAMBDA(const int i, Scalar& ret) { ret += v(i); };
+
+    // This first reduction is necessary to ensure we trigger the bug #8443
+    Scalar ret;
+    Kokkos::parallel_reduce(Kokkos::RangePolicy(0, 1), l,
+                            Kokkos::Sum<Scalar>(ret));
+    EXPECT_EQ(ret, 10);
+
+    // Test that LaunchBounds<1> works
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<Kokkos::LaunchBounds<1>>(0, 1),
+                            l, Kokkos::Sum<Scalar>(ret));
+    EXPECT_EQ(ret, 10);
```

Comment:
I would prefer to also test something where the identity is not given by default/zero initialization, say Kokkos::Prod.